### PR TITLE
[WIP] refactor: Converting TableElement.jsx to functional component

### PR DESCRIPTION
**Pull request by lyndsiWilliams**
### SUMMARY
TableElement.jsx at src/SqlLab/components/TableElement.jsx converted from class to functional component.

### TEST PLAN
Current tests are not passing, they make checks through component state which exists in class components but not in functional components. Below is a screenshot of the TableElement rendering correctly after converting from class to functional component.
Screen Shot 2021-02-05 at 10 46 38 AM

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/SqlLab/components/TableElement.jsx [hunk 1]
```diff
@@ -52,59 +51,40 @@ const StyledSpan = styled.span`
...
-    this.toggleSortColumns = this.toggleSortColumns.bind(this);
-    this.removeTable = this.removeTable.bind(this);
-    this.setHover = this.setHover.bind(this);
-  }
-
-  setHover(hovered) {
-    this.setState({ hovered });
-  }
+function TableElement({ actions, table, timeout }) {
+  const [state, setState] = useState({
```
Comment (lyndsiWilliams) [inline superset-frontend/src/SqlLab/components/TableElement.jsx:55]: Break these into individual hooks

**Comment by lyndsiWilliams** [comment]
Still working on PR summary and testing.

**Comment by eschutho** [comment]
Do you want to set this as a "draft", too?

File: superset-frontend/src/SqlLab/components/TableElement.jsx (hunk 1 above)
Comment (lyndsiWilliams) [reply superset-frontend/src/SqlLab/components/TableElement.jsx]: Completed at [`this commit`](https://github.com/apache/superset/pull/12949/commits/28bb86ab8231a8c03490d6e69c269287fde472d2)

**Comment by lyndsiWilliams** [comment]
Corbin has taken this over at [this PR](https://github.com/apache/superset/pull/14830).

**Pull request closed without merging**